.github/workflows/ci.yml

```diff
@@ -70,6 +70,38 @@ jobs:
         # See tools/ci/src/main.rs for the commands this runs
         run: cargo run -p ci -- nonlocal
 
+  miri:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
```

Comment:
This seems to not cache `~/.cargo/{.crates2.json,.crates.toml}` so I assume it will still rebuild xargo each time? I am not sure how caching works for GHA though.

Reply:
the list of paths to cache is from https://github.com/actions/cache/blob/main/examples.md#rust---cargo... maybe it's not complete?